additional_resources.md

```diff
@@ -11,3 +11,68 @@ A collection of complementary helpful content
 ## Books (Online)
 
 - [The Rust API Guidelines](https://rust-lang.github.io/api-guidelines)
+
+## Design principles
+
+### [SOLID](https://en.wikipedia.org/wiki/SOLID)
+
+- [Single Responsibility Principle (SRP)](https://en.wikipedia.org/wiki/Single-responsibility_principle) -
+  A class should only have a single responsibility, that is, only changes to one part of the software's
+  specification should be able to affect the specification of the class.
+- [Open/Closed Principle (OCP)](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle) -
+  "Software entities ... should be open for extension, but closed for modification."
+- [Liscov Substitution Principle (LSP)](https://en.wikipedia.org/wiki/Liskov_substitution_principle) -
+  "Objects in a program should be replaceable with instances of their subtypes without altering the correctness
+  of that program."
+- [Interface Segregation Principle (ISP)](https://en.wikipedia.org/wiki/Interface_segregation_principle) -
+  "Many client-specific interfaces are better than one general-purpose interface."
+- [Dependency Inversion Principle (DIP)](https://en.wikipedia.org/wiki/Dependency_inversion_principle) -
+  One should "depend upon abstractions, [not] concretions."
+
+### [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+
+- "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```

Comment:
```suggestion
- [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```
I think that like this they are nicer to read

Reply:
`:` gets removed by mdlint

Reply:
but i think it's not necessary to add a lint for this

Reply:
You can use ` - ` or something like that while we investigate this. Anyway, do you like this as a list better?

Reply:
I think as it's now without hyphen or - it's fine, take a look